service/hub/api/authz.go

```diff
@@ -0,0 +1,45 @@
+package api
+
+import (
+	"fmt"
+	"net"
+	"strings"
+)
+
+// Authorizer interface performs authorization of the request.
+// It is invoked after authentication, so it can be assumed that the requestor has
+// presented satisfactory credentials for Account.
+// Principally, it answers the question: is this Account allowed to perform these Actions
+// on this Type.Name subject in the give Service?
+type Authorizer interface {
```

Comment:
`authn` and `authz` means what? `Authenticator` interface and `Authorizer` interface what is the relationship？

Reply:
authn will check user and its password is right or not, authz will range req's scope and check if the user have the right to pull/push image.